From b63c73dc5c37e5405bf032c9113c1daced3e45a4 Mon Sep 17 00:00:00 2001
From: Henrik Gramner <henrik@gramner.com>
Date: Tue, 7 Aug 2018 22:42:22 +0200
Subject: [PATCH] cli: Fix linking with --system-libx264 on x86

---
 common/x86/cpu-a.asm | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/common/x86/cpu-a.asm b/common/x86/cpu-a.asm
index d94f7d54..a41d67f8 100644
--- a/common/x86/cpu-a.asm
+++ b/common/x86/cpu-a.asm
@@ -83,7 +83,7 @@ cglobal cpu_sfence
 ;-----------------------------------------------------------------------------
 ; intptr_t stack_align( void (*func)(void*), ... ); (up to 5 args)
 ;-----------------------------------------------------------------------------
-cglobal stack_align
+cvisible stack_align
     mov      rax, r0mp
     mov       r0, r1mp
     mov       r1, r2mp
@@ -131,7 +131,7 @@ cglobal cpu_cpuid_test
     popfd
     ret
 
-cglobal stack_align
+cvisible stack_align
     push      ebp
     mov       ebp, esp
     sub       esp, 20
-- 
2.11.0

