--- ./configure.orig	2019-03-12 00:41:42.000000000 +0000
+++ ./configure	2019-04-29 03:48:04.806285957 +0100
@@ -1206,7 +1206,7 @@ fi
 
 if [ "$gpac" = "auto" -a "$lsmash" != "yes" ] ; then
     gpac="no"
-    GPAC_LIBS="-lgpac_static"
+    GPAC_LIBS="-lgpac"
     cc_check "" -lz && GPAC_LIBS="$GPAC_LIBS -lz"
     if [ "$SYS" = "WINDOWS" ] ; then
         cc_check "" -lws2_32 && GPAC_LIBS="$GPAC_LIBS -lws2_32"
