diff -up x264-stable-20110714/configure.nover x264-stable-20110714/configure
--- x264-stable-20110714/configure.nover	2011-07-14 19:40:29.000000000 +0200
+++ x264-stable-20110714/configure	2011-07-14 19:42:02.213802421 +0200
@@ -1038,7 +1038,7 @@ fi
 echo "LDFLAGSCLI = $LDFLAGSCLI" >> config.mak
 echo "CLI_LIBX264 = $CLI_LIBX264" >> config.mak
 
-./version.sh >> config.h
+cat version.h >> config.h
 
 pclibs="-L$libdir -lx264 $libpthread"
 
