diff -up x264-20081202/configure.r x264-20081202/configure
--- x264-20081202/configure.r	2008-12-02 23:26:25.000000000 +0100
+++ x264-20081202/configure	2008-12-03 01:19:48.000000000 +0100
@@ -41,7 +41,7 @@ as_check() {
     return $TMP
 }
 
-rm -f config.h config.mak x264.pc conftest*
+rm -f config.mak x264.pc conftest*
 
 prefix='/usr/local'
 exec_prefix='${prefix}'
@@ -239,6 +239,7 @@ case $host_cpu in
   i*86)
     ARCH="X86"
     ASFLAGS="-O2"
+    LDFLAGS="$LDFLAGS -m32"
     if [ "$SYS" = MACOSX ]; then
       ASFLAGS="$ASFLAGS -f macho -DPREFIX"
     elif [ "$SYS" = MINGW ]; then
@@ -352,7 +353,11 @@ if test "$pthread" = "yes" ; then
     LDFLAGS="$LDFLAGS $libpthread"
 fi
 
+if [ "$shared" = "yes" ]; then
+MP4_LDFLAGS="-lgpac"
+else
 MP4_LDFLAGS="-lgpac_static"
+fi
 if [ $SYS = MINGW ]; then
     MP4_LDFLAGS="$MP4_LDFLAGS -lwinmm"
 fi
@@ -452,8 +457,6 @@ if [ "$shared" = "yes" ]; then
     echo 'default: $(SONAME)' >> config.mak
 fi
 
-./version.sh
-
 pclibs="-L$libdir -lx264 $libpthread"
 
 cat > x264.pc << EOF
diff -up x264-20081202/Makefile.r x264-20081202/Makefile
--- x264-20081202/Makefile.r	2008-12-02 23:26:25.000000000 +0100
+++ x264-20081202/Makefile	2008-12-03 01:21:24.000000000 +0100
@@ -53,6 +53,7 @@ ALTIVECSRC += common/ppc/mc.c common/ppc
               common/ppc/predict.c
 SRCS += $(ALTIVECSRC)
 $(ALTIVECSRC:%.c=%.o): CFLAGS += $(ALTIVECFLAGS)
+common/ppc/pixel.c:    CFLAGS += -flax-vector-conversions
 endif
 
 # VIS optims
@@ -77,14 +78,25 @@ libx264.a: .depend $(OBJS) $(OBJASM)
 	ar rc libx264.a $(OBJS) $(OBJASM)
 	ranlib libx264.a
 
+libx264.so: $(SONAME)
+	ln -sf $(SONAME) libx264.$(SOSUFFIX)
+
 $(SONAME): .depend $(OBJS) $(OBJASM)
 	$(CC) -shared -o $@ $(OBJS) $(OBJASM) $(SOFLAGS) $(LDFLAGS)
 
+ifneq ($(SONAME),)
+x264$(EXE): $(OBJCLI) libx264.so
+	$(CC) -o $@ $+ $(LDFLAGS) -lx264 -L.
+
+checkasm: tools/checkasm.o $(SONAME)
+	$(CC) -o $@ $+ $(LDFLAGS) -lx264 -L.
+else
 x264$(EXE): $(OBJCLI) libx264.a 
 	$(CC) -o $@ $+ $(LDFLAGS)
 
 checkasm: tools/checkasm.o libx264.a
 	$(CC) -o $@ $+ $(LDFLAGS)
+endif
 
 %.o: %.asm
 	$(AS) $(ASFLAGS) -o $@ $<
@@ -148,10 +160,12 @@ install: x264$(EXE) $(SONAME)
 	install -d $(DESTDIR)$(bindir) $(DESTDIR)$(includedir)
 	install -d $(DESTDIR)$(libdir) $(DESTDIR)$(libdir)/pkgconfig
 	install -m 644 x264.h $(DESTDIR)$(includedir)
+ifeq ($(SONAME),)
 	install -m 644 libx264.a $(DESTDIR)$(libdir)
+	ranlib $(DESTDIR)$(libdir)/libx264.a
+endif
 	install -m 644 x264.pc $(DESTDIR)$(libdir)/pkgconfig
 	install x264$(EXE) $(DESTDIR)$(bindir)
-	ranlib $(DESTDIR)$(libdir)/libx264.a
 ifeq ($(SYS),MINGW)
 	$(if $(SONAME), install -m 755 $(SONAME) $(DESTDIR)$(bindir))
 else
