--- ./configure.orig	2020-11-22 22:49:29.765327990 +0000
+++ ./configure	2020-11-23 01:07:38.050590012 +0000
@@ -1244,9 +1244,7 @@ fi
 
 if [ "$gpac" = "auto" -a "$lsmash" != "yes" ] ; then
     gpac="no"
-    GPAC_LIBS="-lgpac_static"
-    cc_check "" -lz && GPAC_LIBS="$GPAC_LIBS -lz"
-    cc_check "" -ldl && GPAC_LIBS="$GPAC_LIBS -ldl"
+    GPAC_LIBS="$($PKGCONFIG  --static --libs gpac)"
     if [ "$SYS" = "WINDOWS" ] ; then
         cc_check "" -lws2_32 && GPAC_LIBS="$GPAC_LIBS -lws2_32"
         cc_check "" -lwinmm && GPAC_LIBS="$GPAC_LIBS -lwinmm"
