diff -up x264-20100116/configure.nostrip x264-20100116/configure
--- x264-20100116/configure.nostrip	2010-01-16 13:19:59.000000000 +0100
+++ x264-20100116/configure	2010-01-16 13:40:39.000000000 +0100
@@ -491,6 +491,8 @@ fi
 if [ "$debug" != "yes" -a "$gprof" != "yes" ]; then
     CFLAGS="$CFLAGS -s -fomit-frame-pointer"
     LDFLAGS="$LDFLAGS -s"
+else
+    STRIP="/bin/true"
 fi
 
 if [ "$debug" = "yes" ]; then
